plugins/slo/pkg/slo/ir.go

```diff
@@ -0,0 +1,216 @@
+package slo
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"regexp"
+	"strings"
+	"time"
+
+	openslov1 "github.com/alexandreLamarre/oslo/pkg/manifest/v1"
+	"github.com/alexandreLamarre/sloth/core/app/generate"
+	"github.com/alexandreLamarre/sloth/core/info"
+	"github.com/alexandreLamarre/sloth/core/prometheus"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	specTypeV1AlphaRegexKind       = regexp.MustCompile(`(?m)^kind: +['"]?SLO['"]? *$`)
+	specTypeV1AlphaRegexAPIVersion = regexp.MustCompile(`(?m)^apiVersion: +['"]?openslo\/v1alpha['"]? *$`)
+)
+
+var errorRatioRawQueryTpl = template.Must(template.New("").Parse(`
+  1 - (
+    (
+      {{ .good }}
+    )
+    /
+    (
+      {{ .total }}
+    )
+  )
+`))
```

Comment:
For the templates in other files:
```
goodQueryTempl = template.Must(template.New("").Parse(`
		sum(rate({{.metricId}}{job="{{.jobId}}", {{.filter}}}[{{"{{.window}}"}}]))
	`))
```
we trim all the whitespace , \n and \t from the `Execute` result using `TrimWhitespaces` which gives:

`sum(rate(up{job="foo-service", up=1}[{{.window}}]))`,which should be ok?

But for the one in `ir.go` I'm following the sloth internals from `core/spec.go` which has the same format. A little hesitant to change this one since the sloth internals are kinda wonky